subsys/net/ip/utils.c

```diff
@@ -518,3 +518,230 @@ bool net_header_fits(struct net_pkt *pkt, u8_t *hdr, size_t hdr_size)
 
 	return false;
 }
+
+#if defined(CONFIG_NET_IPV6) || defined(CONFIG_NET_IPV4)
+static bool convert_port(const char *buf, u16_t *port)
+{
+	char *endptr;
+	int tmp;
+
+	tmp = strtol(buf, &endptr, 10);
+	if ((endptr == buf && tmp == 0) ||
+	    !(*buf != '\0' && *endptr == '\0') ||
+	    tmp > 0xffff || tmp < 0) {
+		return false;
+	}
+
+	*port = tmp;
+
+	return true;
+}
+#endif /* CONFIG_NET_IPV6 || CONFIG_NET_IPV4 */
+
+bool net_ipaddr_parse(const char *str, size_t str_len, struct sockaddr *addr)
+{
+	int count, i, ret;
+	u16_t port;
+	char *ptr;
+
+	if (*str == '[') {
+#if defined(CONFIG_NET_IPV6)
+		/* IPv6 address with port number */
+		struct in6_addr *addr6;
+		char ipaddr[INET6_ADDRSTRLEN + 1];
+		int end, len;
+
+		ptr = strstr(str, "]");
```

Comment:
Please use `memchr()` here: since there's a `str_len` parameter, a user might pass a non-zero-delimited string.  Moreover, `strstr()` is overkill to scan for a single character.